catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveTableOperations.java

```diff
@@ -215,8 +222,107 @@ private org.apache.hadoop.hive.metastore.api.Partition toHivePartition(
+          table
+              .clientPool()
+              .run(
+                  c ->
+                      c.listPartitions(
+                          table.schemaName(),
+                          table.name(),
+                          getFilterPartitionValueList(hiveTable, partitionName),
+                          (short) -1));
+      if (partitions.isEmpty()) {
+        throw new NoSuchPartitionException(
+            "Hive partition %s does not exist in Hive Metastore", partitionName);
+      }
+      // Delete partitions iteratively
+      for (org.apache.hadoop.hive.metastore.api.Partition partition : partitions) {
+        table
+            .clientPool()
+            .run(
+                c ->
+                    c.dropPartition(
+                        partition.getDbName(),
+                        partition.getTableName(),
+                        partition.getValues(),
+                        false));
+      }
+    } catch (NoSuchPartitionException e) {
+      if (ifExists) {
+        return true;
+      }
+      throw new NoSuchPartitionException(
+          "Hive partition %s does not exist in Hive Metastore", partitionName);
+
+    } catch (UnknownTableException e) {
+      throw new NoSuchTableException(
+          e, "Hive table %s does not exist in Hive Metastore", table.name());
+
+    } catch (TException | InterruptedException e) {
+      throw new RuntimeException(
+          "Failed to get partition "
+              + partitionName
+              + " of table "
+              + table.name()
+              + "from Hive Metastore",
+          e);
+    }
+    return true;
+  }
+
+  /**
+   * Convert partition format [String -> Array], to get the query criteria of hmsClient.listPartitions()
+   * Example : for a three-level partitioned table "log_date=xxxx/log_hour=xx/log_hour=xx"
+   * 1. input partition: "log_date=0101/log_hour=01/log_hour=02"
+   * Query criteria: ["0101","01","02"], hmsClient.listPartitions() will return the partition detail of "log_date=0101/log_hour=01/log_hour=02"
+   * 2. input partition: "log_date=0101"
+   * Query criteria: ["0101","",""], hmsClient.listPartitions() will return the detail of all partition that contains "log_date=0101"
+   * @param dropTable table details
+   * @param partitionSpec partition in String format
+   * @return the filter partition list
+   * @throws NoSuchPartitionException
+   */
+  private List<String> getFilterPartitionValueList(Table dropTable, String partitionSpec)
+      throws NoSuchPartitionException {
+    Map<String, String> partMap = new HashMap<>();
+    String[] parts = partitionSpec.split(PARTITION_NAME_DELIMITER);
+    for (String part : parts) {
+      String[] keyValue = part.split(PARTITION_VALUE_DELIMITER);
+      if (keyValue.length == 2) {
+        partMap.put(keyValue[0], keyValue[1]);
+      } else {
+        LOG.error("Error partition format: {}", partitionSpec);
+        throw new NoSuchPartitionException(
```

Comment:
Here should throw `IllegalArgumentException`

Reply:
done